### PR TITLE
Call `Unalias()` in `IsNilable()` to support `gotypesalias=1`

### LINKED ISSUE #3301: Bug found when generating schema including arrays of type Any
### What happened?

We have a schema that uses an array of type Any as one of the field's types. Starting with `v0.17.48`, when we `go run github.com/99designs/gqlgen generate`, we encounter a breaking error that essentially adds an extra pointer when unmarshaling: Screenshot 2024-09-25 at 3 12 48 PM

### What did you expect?

Code generation to continue to work even when including a type of `[Any]` in the schema

### Minimal graphql.schema and models to reproduce

```graphqls
scalar Any

type Broken {
   fakeField: [Any]!
}
```

### versions
 - `gqlgen >=v0.17.48`
 - `go v1.23.1`

### PR DESCRIPTION
**Pull request by noamcohen97**
Fixed false `IsNilable()` value for any (or any other aliased type) when `gotypesalias=1` (default since Go 1.23.0)
Fixes #3301

I have:
 - [x] Added tests covering the bug / feature (see [testing](https://github.com/99designs/gqlgen/blob/master/TESTING.md))
 - [ ] Updated any relevant documentation (see [docs](https://github.com/99designs/gqlgen/tree/master/docs/content))

### CODE REVIEW

File: codegen/config/binder.go [hunk 1]
```diff
@@ -520,6 +520,7 @@ func (b *Binder) CopyModifiersFromAst(t *ast.Type, base types.Type) types.Type {
 }
 
 func IsNilable(t types.Type) bool {
+	t = code.Unalias(t)
```
Comment (noamcohen97) [inline codegen/config/binder.go:523]: Maybe we should call `types.Unalias(t)` so we'll Unalias even in Go 1.22?
Currently in Go 1.22 calling this with a nilable alias type returns false.
@StevenACoffman, @a8m WDYT?

Comment (StevenACoffman) [reply codegen/config/binder.go:523]: We really need to continue to support Go 1.22, and I don't think we can just assume people have gotypesalias=1

Comment (noamcohen97) [reply codegen/config/binder.go]: So I'll call `types.Unalias()` here, that way we'll make sure `IsNilable()` is reliable no matter what.
Also - we don't need the pointer base type unaliasing from `code.Unalias()` here.

**Comment by StevenACoffman** [comment]
Thanks! I really appreciate your fix here! Looking forward to your next contribution!

**Comment by noamcohen97** [comment]
Thanks @StevenACoffman!
When can we expect a new release?

**Comment by noamcohen97** [comment]
Thanks @StevenACoffman!
When can we expect a new release?

**Comment by StevenACoffman** [comment]
Right now! Just released!

**Pull request merged**